data_analysis/near-deduplication/README.md

```diff
@@ -2,29 +2,211 @@
 
 Code for data near deduplication.
 
-### Setup 
+### Setup
+
 ````
 pip install -r requirements.txt
 ````
 
 Login to be able to be able to push the dataset to the hub after deduplication and clone your huggingface-hub repositories:
+
 ````
 huggingface-cli login
 ````
+
 And make sure you have git-lfs installed.
 
-If you use datasets with different column names from the BigCode ones, you might need to change `PATH_COLUMN` and `CONTENT` variables in `minhash_deduplication.py`. 
+If you use datasets with different column names from the BigCode ones, you might need to change `PATH_COLUMN` and `CONTENT` variables in `minhash_deduplication.py`.
+
+### Usage
 
-### Usage 
 To run near deduplication use the following command and adapt the arguments for your case:
+
 ````
 python near_deduplicate.py \
     --dataset_name bigcode-data/python_any_license_v2 \
     --org bigcode-data \
     --repo_name python_any_license_v2_near_dedup \
     --out_path ./data/any_license-near-dedup \
     --text_column content 
-```` 
-To make just a test run with a subset of the data set `test_run` argument to True. 
+````
+
+To make just a test run with a subset of the data set `test_run` argument to True.
 
 The first time you load the dataset might be slow if it is large, but the data is saved in the cache thanks to `datasets`, and the subsequent calls will be fast.
+
+### Alternative Deduplication Script
+
+`minhash_deduplication_alt.py` is an alternative you might find useful to use as well. It uses `dataset.map` for both hashing and querying as well as `networkx` for duplicate community detection.
+
+I will add an arg parser later, but for now you can update the config in `minhash_deduplication_alt.py` to use the script on custom dataset.
+
+```bash
+pip install -r requirements_alt.txt
+python minhash_deduplication_alt.py
+```
+
+#### Benchmark
+
+Running this script for `codeparrot/codeparrot-clean` on a 10-core machine:
+
+```plaintext
+Original size: 5361373
+Removed size: 2077902
+Processing time taken: 3567.90 seconds
+```
+
+```plaintext
+                                                                                   Some examples of duplicate code                                                                                 
```

Comment:
Maybe we can store these examples in a separate folder or in a dataset on the hub to not have a very large README

Reply:
I will remove them since people can easily get that by running the script.